tutorials/example_pytorch_qat.py

```diff
@@ -0,0 +1,223 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""
+This tutorial demonstrates how to perform Quantization Aware Training (QAT) using the Model Compression Toolkit (MCT).
+We first train a simple model on MNIST dataset, then quantize the model and transform it to a QAT-ready model.
+A QAT-ready model is a model in which certain layers are wrapped by "quantization wrappers" with requested quantizers.
+The user can now Fine-Tune the QAT-ready model. Finally, the model is finalized by the MCT which means the
+MCT replaces the "quantization wrappers" with their native layers and quantized weights.
+"""
+
+
+from __future__ import print_function
+import argparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+import model_compression_toolkit as mct
+
+
+def argument_handler():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--representative_dataset_dir', type=str, required=True, default=None,
+                        help='folder path for the representative dataset.')
+    parser.add_argument('--batch_size', type=int, default=64,
+                        help='batch size for the representative data.')
+    parser.add_argument('--num_calibration_iterations', type=int, default=10,
+                        help='number of iterations for calibration.')
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+
+    # Parse arguments
+    args = argument_handler()
+
+    # Let us define the network and some helper functions to train and evaluate the model.
+    # These are taken from the official Pytorch examples https://github.com/pytorch/examples/blob/main/mnist/main.py
+    class Net(nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.conv1 = nn.Conv2d(1, 32, 3, 1)
+            self.conv2 = nn.Conv2d(32, 64, 3, 1)
+            self.dropout1 = nn.Dropout(0.25)
+            self.dropout2 = nn.Dropout(0.5)
+            self.fc1 = nn.Linear(9216, 128)
+            self.fc2 = nn.Linear(128, 10)
+
+        def forward(self, x):
+            x = self.conv1(x)
+            x = F.relu(x)
+            x = self.conv2(x)
+            x = F.relu(x)
+            x = F.max_pool2d(x, 2)
+            x = self.dropout1(x)
+            x = torch.flatten(x, 1)
+            x = self.fc1(x)
+            x = F.relu(x)
+            x = self.dropout2(x)
+            x = self.fc2(x)
+            output = F.log_softmax(x, dim=1)
+            return output
+
+    # training function
```

Comment:
It's really petty an not important, but some of the comments start with a Capital letter and some aren't, please make it consistent.